sdks/python/apache_beam/io/gcp/bigtableio.py

```diff
@@ -36,43 +36,170 @@
+
+  def split(self, desired_bundle_size=None, start_position=None, stop_position=None):
+    """ Splits the source into a set of bundles, using the row_set if it is set.
+
+    *** At this point, only splitting an entire table into samples based on the sample row keys is supported ***
+
+    :param desired_bundle_size: the desired size (in bytes) of the bundles returned.
+    :param start_position: if specified, the position must be used as the starting position of the first bundle.
+    :param stop_position: if specified, the position must be used as the ending position of the last bundle.
+    Returns:
+    	an iterator of objects of type 'SourceBundle' that gives information about the generated bundles.
+    """
+
+    if desired_bundle_size is not None or start_position is not None or stop_position is not None:
+      raise NotImplementedError
+
+    # TODO: Use the desired bundle size to split accordingly
+    # TODO: Allow users to provide their own row sets
+
+    sample_row_keys = list(self.get_sample_row_keys())
+    bundles = []
+    if len(sample_row_keys) > 0 and sample_row_keys[0] != b'':
+        bundles.append(SourceBundle(sample_row_keys[0].offset_bytes, self, b'', sample_row_keys[0].row_key))
+    for i in range(1, len(sample_row_keys)):
+      pos_start = sample_row_keys[i - 1].offset_bytes
+      pos_stop = sample_row_keys[i].offset_bytes
+      bundles.append(SourceBundle(pos_stop - pos_start, self,
+                                  sample_row_keys[i - 1].row_key,
+                                  sample_row_keys[i].row_key))
+
+    # Shuffle is needed to allow reading from different locations of the table for better efficiency
+    shuffle(bundles)
+    return bundles
+
+  def read(self, range_tracker):
+    for row in self._get_table().read_rows(start_key=range_tracker.start_position(),
+                                           end_key=range_tracker.stop_position(),
+                                           filter_=self.beam_options['filter_']):
+      if range_tracker.try_claim(row.row_key):
+        self.row_count.inc()
+        yield row
+      else:
+        # TODO: Modify the client ot be able to cancel read_row request
+        break
+
+  def display_data(self):
+    ret = {'projectId': DisplayDataItem(self.beam_options['project_id'],
+                                        label='Bigtable Project Id',
+                                        key='projectId'),
+           'instanceId': DisplayDataItem(self.beam_options['instance_id'],
+                                         label='Bigtable Instance Id',
+                                         key='instanceId'),
+           'tableId': DisplayDataItem(self.beam_options['table_id'],
+                                      label='Bigtable Table Id',
+                                      key='tableId')}
+    return ret
+
+  def to_runner_api_parameter(self, unused_context):
+    pass
+
+
+class _BigtableWriteFn(DoFn):
```

Comment:
Please revert any non-read related changes.  Those changes should go in a separate PR.